### PR TITLE
cni: fix on GKE with oslogin

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/48416

Please read the issue, it has a lots of details on this bizarre case.

### CODE REVIEW

File: tools/istio-iptables/pkg/dependencies/implementation_linux.go [hunk 1]
```diff
@@ -104,18 +104,32 @@ func (r *RealDependencies) executeXTables(cmd string, ignoreErrors bool, stdin i
 			// In both cases we are setting the lockfile to `r.NetworkNamespace`.
 			// * /dev/null looks like a good option, but actually doesn't work (it will ensure only one actor can access it)
 			// * `mktemp` works, but it is slightly harder to deal with cleanup and in some platforms we may not have write access.
+			//
+			// In some setups, iptables can make remote network calls(!!). Since these come from a partially initialized pod network namespace,
```
Comment (zirain) [inline tools/istio-iptables/pkg/dependencies/implementation_linux.go:108]: TIL

**Comment by edtanous** [comment]
LGTM.  A little bit odd, but does seem to solve a problem.

**Comment by briansonnenberg** [comment]
I don't see any practical issues with using mount bind here, so this LGTM.

**Review by GregHanson — APPROVED** [review]
lgtm

**Pull request merged**